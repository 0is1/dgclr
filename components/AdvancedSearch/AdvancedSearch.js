// @flow
import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import Router from 'next/router';
import { Box } from 'rebass';
import type { State as ReduxState } from 'lib/types';
import AdvancedSearchInputsComponent from './AdvancedSearchInputs';
import AdvancedSearchQueryComponent from './AdvancedSearchQuery';
import { getCurrentAdvancedFilter, isAdvancedSearchMapVisible } from './selectors';
import Styles from './AdvancedSearch.styles';

const { Wrapper } = Styles;
const FILTER_UPDATED = 'FILTER_UPDATED';

type Props = {};
type MapStateToProps = { filter: string, mapChecked: boolean };
type MapDispatchToProps = {};

type CombinedProps = Props & MapStateToProps & MapDispatchToProps;

export class AdvancedSearch extends PureComponent<CombinedProps> {
  componentDidMount() {
    const { query } = Router;
    const { filter } = this.props;
    if (query && Object.keys(query).length === 0 && filter) {
      this.updateUrlQuery();
    }
  }

  componentDidUpdate(prevProps: CombinedProps, prevState: any, snapshot: any) {
    if (snapshot === FILTER_UPDATED) {
      this.updateUrlQuery();
    }
  }

  getSnapshotBeforeUpdate(prevProps: CombinedProps) {
    const { filter, mapChecked } = this.props;
    if (prevProps.filter !== filter || mapChecked !== prevProps.mapChecked) {
      return FILTER_UPDATED;
    }
    return null;
  }

  updateUrlQuery = () => {
    const { pathname } = Router;
    const { filter, mapChecked } = this.props;
    const filterObject = JSON.parse(filter);
    const urlFilter = { ...filterObject, mapChecked };
    const href = `${pathname}?q=${encodeURIComponent(JSON.stringify(urlFilter))}`;
    Router.push(href, href, { shallow: true });
  };

  render() {
    const { filter } = this.props;
    return (
      <Wrapper>
        <Box style={{ position: 'relative' }} m="1rem auto" px="2rem" width={[1, 1, 1, 0.7]}>
          <AdvancedSearchInputsComponent />
          <AdvancedSearchQueryComponent filter={JSON.parse(filter)} />
        </Box>
      </Wrapper>
    );
  }
}

const mapStateToProps = (state: ReduxState): MapStateToProps => ({
  mapChecked: isAdvancedSearchMapVisible(state),
  filter: getCurrentAdvancedFilter(state),
});

export default connect<CombinedProps, Props, any, any, any, Function>(mapStateToProps)(AdvancedSearch);
