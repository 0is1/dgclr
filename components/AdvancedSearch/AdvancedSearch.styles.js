import styled from 'styled-components';

const Styles = {
  Wrapper: styled.div`
    margin: 0 auto;
    padding-top: 1rem;
  `,
  ToggleButtonWrapper: styled.div`
    padding: 0.25rem;
    position: absolute;
    right: 0;
    top: 0;
    width: 30px;
    &:hover {
      cursor: pointer;
    }
  `,
};

export default Styles;
