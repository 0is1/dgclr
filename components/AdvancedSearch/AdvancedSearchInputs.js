// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  Box, Flex, Label, Text,
} from 'rebass';
import update from 'updeep';
import { omit, size } from 'lodash';

import RatingSelect from 'components/Select/RatingSelect';
import BasketTypeSelect from 'components/Select/BasketTypeSelect';
import TeeTypeSelect from 'components/Select/TeeTypeSelect';
import SurfaceTypeSelect from 'components/Select/SurfaceTypeSelect';
import CourseTypeSelect from 'components/Select/CourseTypeSelect';
import AdvancedSearchMap from 'components/Map/AdvancedSearchMap';
import Toggle from 'components/Toggle';
import colors from 'components/colors';
import Input from 'components/Input';
import type { CoordinatesObject, State } from 'lib/types';
import { convertMetersToKilometers } from 'helpers/utils';
import { ADVANCED_NEARBY, ADVANCED_COURSE_INFO } from 'lib/constants';
import { INPUT_FILTER_NAMES } from 'components/Input/constants';
import { SELECT_FILTER_NAMES } from 'components/Select/constants';
import RebassComponents from 'components/RebassComponents';
import {
  getCurrentAdvancedFilter,
  isAdvancedSearchMapVisible,
  isAllAdvancedSearchInputsOpen,
} from './selectors';
import {
  setCurrentAdvancedSearchFilter,
  toggleAdvancedSearchMap,
  toggleAdvancedSearchInputs as toggleAdvancedSearchInputsFunc,
} from './actions';
import Styles from './AdvancedSearchInputs.styles';

const { Divider } = RebassComponents;
const { FadeInBox } = Styles;

type Props = {};
type MapStateToProps = {
  filter: string,
  mapChecked: boolean,
  allInputsOpen: boolean,
};

type MapDispatchToProps = {
  setAdvancedSearchFilter: Function,
  toggleMapVisibility: Function,
  toggleAdvancedSearchInputs: Function,
};

type CombinedProps = Props & MapStateToProps & MapDispatchToProps;
type FilterType = { courseInfo: {} };

export class AdvancedSearchInputs extends Component<CombinedProps> {
  getParsedFilter = () => {
    const { filter } = this.props;
    return !filter.length ? {} : JSON.parse(filter);
  };

  setAdvancedSearchFilterData = (newFilter: FilterType) => {
    const { setAdvancedSearchFilter } = this.props;
    // console.log('newFilter: ', this.cleanCourseInfo(newFilter));
    setAdvancedSearchFilter(JSON.stringify(this.cleanCourseInfo(newFilter)));
  };

  cleanCourseInfo = (newFilter: FilterType) => {
    if (newFilter.courseInfo && size(newFilter.courseInfo) === 0) return omit(newFilter, [ADVANCED_COURSE_INFO]);
    return newFilter;
  };

  onRatingChange = (rating: []) => {
    // console.log('onRatingChange: ', rating);
    const filter = this.getParsedFilter();
    // console.log('onRatingChange filter: ', filterData);
    const newFilter = rating.length > 0
      ? update({ rating }, filter)
      : omit(filter, [SELECT_FILTER_NAMES.rating.filterName]);
    this.setAdvancedSearchFilterData(newFilter);
  };

  onBasketTypeChange = (basketType: string) => {
    const filter = this.getParsedFilter();
    // console.log('onBasketTypeChange filter: ', filter);
    const newFilter = basketType.length > 0
      ? update({ courseInfo: { basketType } }, filter)
      : {
        ...filter,
        courseInfo: omit(filter.courseInfo, [
          SELECT_FILTER_NAMES.basketType.filterName,
        ]),
      };
    this.setAdvancedSearchFilterData(newFilter);
  };

  onTeeTypeChange = (teeType: string) => {
    const filter = this.getParsedFilter();
    // console.log('onTeeTypeChange filter: ', filter);
    const newFilter = teeType.length > 0
      ? update({ courseInfo: { teeType } }, filter)
      : {
        ...filter,
        courseInfo: omit(filter.courseInfo, [
          SELECT_FILTER_NAMES.teeType.filterName,
        ]),
      };
    this.setAdvancedSearchFilterData(newFilter);
  };

  onSurfaceTypeChange = (surfaceShapeTypes: string) => {
    const filter = this.getParsedFilter();
    // console.log('surfaceType filter: ', filter);
    const newFilter = surfaceShapeTypes.length > 0
      ? update({ courseInfo: { surfaceShapeTypes } }, filter)
      : {
        ...filter,
        courseInfo: omit(filter.courseInfo, [
          SELECT_FILTER_NAMES.surfaceShapeTypes.filterName,
        ]),
      };
    this.setAdvancedSearchFilterData(newFilter);
  };

  onCourseTypeChange = (courseTypes: string) => {
    const filter = this.getParsedFilter();
    // console.log('courseTypes filter: ', filter);
    const newFilter = courseTypes.length > 0
      ? update({ courseInfo: { courseTypes } }, filter)
      : {
        ...filter,
        courseInfo: omit(filter.courseInfo, [
          SELECT_FILTER_NAMES.courseTypes.filterName,
        ]),
      };
    this.setAdvancedSearchFilterData(newFilter);
  };

  onMapSearchChange = (data: {
    coordinates: CoordinatesObject,
    radius: number,
  }) => {
    const filter = this.getParsedFilter();
    //  list the longitude first and then latitude https://docs.mongodb.com/manual/reference/geojson/#geojson-point
    const nearby = data.coordinates
      && data.radius && {
      maxDistance: convertMetersToKilometers(parseInt(data.radius, 10)),
      coordinates: [data.coordinates.lng, data.coordinates.lat],
    };
    if (nearby) {
      this.setAdvancedSearchFilterData(update({ nearby }, filter));
      return;
    }
    this.omitMapFilter();
  };

  omitMapFilter = () => {
    const filter = this.getParsedFilter();
    this.setAdvancedSearchFilterData(omit(filter, [ADVANCED_NEARBY]));
  };

  handleMapToggle = (mapChecked: boolean) => {
    if (!mapChecked) this.omitMapFilter();
    const { toggleMapVisibility } = this.props;
    toggleMapVisibility(mapChecked);
  };

  toggleAdvancedSearchInputs = () => {
    const { allInputsOpen, toggleAdvancedSearchInputs } = this.props;
    toggleAdvancedSearchInputs(!allInputsOpen);
  };

  onHoleAverageLengthChange = (values: Array<number>) => {
    const filter = this.getParsedFilter();
    const [min, max] = values;
    const newFilter = update(
      { [INPUT_FILTER_NAMES.holeAverageLength.filterName]: { min, max } },
      filter,
    );
    this.setAdvancedSearchFilterData(newFilter);
  };

  render() {
    const { allInputsOpen, mapChecked } = this.props;
    return (
      <Box>
        <Text fontWeight="700" my={2} width="100%">
          Edistynyt haku (varhaisessa kehitysvaiheessa)
        </Text>
        <Flex flexWrap="wrap">
          <Box
            pr={[0, 0, '.5rem', '.5rem']}
            mb=".75rem"
            width={[1, 1, 1 / 2, 1 / 2]}
          >
            <RatingSelect onChange={this.onRatingChange} />
          </Box>
          <Box
            pr={[0, 0, '.5rem', '.5rem']}
            mb=".75rem"
            width={[1, 1, 1 / 2, 1 / 2]}
          >
            <CourseTypeSelect onChange={this.onCourseTypeChange} />
          </Box>
          <Box
            pr={[0, 0, '.5rem', '.5rem']}
            mb=".75rem"
            width={[1, 1, 1 / 2, 1 / 2]}
          >
            <Label>Väylien keskipituus:</Label>
            <Input
              onChange={this.onHoleAverageLengthChange}
              options={{
                type: 'slider',
                initialValues: [40, 180],
                domain: [10, 180],
                step: 10,
                filterName: INPUT_FILTER_NAMES.holeAverageLength.filterName,
              }}
            />
          </Box>
          <FadeInBox show={allInputsOpen}>
            {allInputsOpen && (
              <>
                <Box
                  pr={[0, 0, '.5rem', '.5rem']}
                  mb=".75rem"
                  width={[1, 1, 1 / 2, 1 / 2]}
                >
                  <BasketTypeSelect onChange={this.onBasketTypeChange} />
                </Box>
                <Box
                  pr={[0, 0, '.5rem', '.5rem']}
                  mb=".75rem"
                  width={[1, 1, 1 / 2, 1 / 2]}
                >
                  <TeeTypeSelect onChange={this.onTeeTypeChange} />
                </Box>
                <Box
                  pr={[0, 0, '.5rem', '.5rem']}
                  mb=".75rem"
                  width={[1, 1, 1 / 2, 1 / 2]}
                >
                  <SurfaceTypeSelect onChange={this.onSurfaceTypeChange} />
                </Box>
              </>
            )}
          </FadeInBox>
          <Toggle
            label="Näytä kaikki valinnat:"
            checked={allInputsOpen}
            handleOnChange={this.toggleAdvancedSearchInputs}
          />
          <Box mb=".75rem" width={[1]}>
            <Toggle
              label="Käytä karttahakua:"
              checked={mapChecked}
              handleOnChange={this.handleMapToggle}
            />
            <AdvancedSearchMap
              mapVisible={mapChecked}
              handleChange={this.onMapSearchChange}
            />
          </Box>
        </Flex>
        <Divider w={1} borderColor={colors.info} />
      </Box>
    );
  }
}

const mapStateToProps = (state: State): MapStateToProps => ({
  allInputsOpen: isAllAdvancedSearchInputsOpen(state),
  filter: getCurrentAdvancedFilter(state),
  mapChecked: isAdvancedSearchMapVisible(state),
});
const mapDispatchToProps = (dispatch: Function): MapDispatchToProps => ({
  setAdvancedSearchFilter: filter => dispatch(setCurrentAdvancedSearchFilter(filter)),
  toggleMapVisibility: (visible: boolean) => dispatch(toggleAdvancedSearchMap(visible)),
  toggleAdvancedSearchInputs: (visible: boolean) => dispatch(toggleAdvancedSearchInputsFunc(visible)),
});

export default connect<CombinedProps, Props, any, any, any, Function>(
  mapStateToProps,
  mapDispatchToProps,
)(AdvancedSearchInputs);
