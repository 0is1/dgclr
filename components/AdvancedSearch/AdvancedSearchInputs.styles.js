import styled from 'styled-components';

const Styles = {
  FadeInBox: styled.div`
    height: ${(props) => (props.show ? 'auto' : '0px')};
    width: ${(props) => (props.show ? '100%' : '0px')};
    position: ${(props) => (props.show ? 'inherit' : 'absolute')};
    margin-top: ${(props) => (props.show ? '0px' : '-9999px')};
    opacity: ${(props) => (props.show ? 1 : 0)};
    top: ${(props) => (props.show ? '0px' : '-9999px')};
    transition: opacity 0.5s ease-in-out;
    display: flex;
    flex-wrap: wrap;
  `,
};

export default Styles;
