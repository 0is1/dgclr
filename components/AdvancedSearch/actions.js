// @flow
import type {
  Course,
  FilterData,
  MapFilterData,
  SliderFilterData,
} from 'lib/types';

export const actionTypes = {
  SET_ADVANCED_SEARCH_QUERY: 'dgclr/SET_ADVANCED_SEARCH_QUERY',
  SET_ADVANCED_FILTER_TYPE: 'dgclr/SET_ADVANCED_FILTER_TYPE',
  SET_CURRENT_ADVANCED_FILTER: 'dgclr/SET_CURRENT_ADVANCED_FILTER',
  TOGGLE_ADVANCED_SEARCH_MAP: 'dgclr/TOGGLE_ADVANCED_SEARCH_MAP',
  TOGGLE_ADVANCED_SEARCH_INPUTS: 'dgclr/TOGGLE_ADVANCED_SEARCH_INPUTS',
  SET_ADVANCED_MAP_ZOOM: 'dgclr/SET_ADVANCED_MAP_ZOOM',
};

/**
 * Set filter data to store
 * @param {string} filterName
 * @param {Array} data
 * @return {Object} action of SET_ADVANCED_BASKET_TYPE_FILTER
 */
export const setFilter = (
  filterName: string,
  data: FilterData | MapFilterData | SliderFilterData,
) => ({
  type: actionTypes.SET_ADVANCED_FILTER_TYPE,
  filterName,
  data,
});

/**
 * Set advanced search query action
 * @param {Array} courses
 * @param {string} query
 * @return {Object} action of SET_ADVANCED_SEARCH_QUERY
 */
export const setAdvancedSearchQuery = (
  courses: Array<?Course>,
  query: string,
) => ({
  type: actionTypes.SET_ADVANCED_SEARCH_QUERY,
  courses,
  query,
});

/**
 * Set current advanced search filters
 * @param {string} filter stringified JSON
 * @return {Object} action of SET_CURRENT_ADVANCED_FILTER
 */
export const setCurrentAdvancedSearchFilter = (filter: string = '') => ({
  type: actionTypes.SET_CURRENT_ADVANCED_FILTER,
  filter,
});

/**
 * Toggle advanced search map visibility
 * @param {boolean} visible
 * @return {Object} action of TOGGLE_ADVANCED_SEARCH_MAP
 */
export const toggleAdvancedSearchMap = (visible: boolean) => ({
  type: actionTypes.TOGGLE_ADVANCED_SEARCH_MAP,
  visible,
});

/**
 * Set advanced search map zoom
 * @param {number} zoom
 * @return {Object} action of SET_ADVANCED_MAP_ZOOM
 */
export const setAdvancedSearchMapZoom = (zoom: number) => ({
  type: actionTypes.SET_ADVANCED_MAP_ZOOM,
  zoom,
});

/**
 * Toggle advanced search open
 * @param {boolean} open
 * @return {Object} action of TOGGLE_ADVANCED_SEARCH_INPUTS
 */
export const toggleAdvancedSearchInputs = (open: boolean = false) => ({
  type: actionTypes.TOGGLE_ADVANCED_SEARCH_INPUTS,
  open,
});
