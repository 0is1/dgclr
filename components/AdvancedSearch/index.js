// @flow
import AdvancedSearchComponent from './AdvancedSearch';

export default AdvancedSearchComponent;
