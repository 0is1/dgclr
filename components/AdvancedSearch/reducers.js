import u from 'updeep';
import { createReducer } from 'lib/createReducer';
import { MAP_DEFAULT_ZOOM } from 'lib/constants';
import { actionTypes } from './actions';

export const initialState = {
  allAdvancedSearchInputsOpen: false,
  queries: {},
  advancedQueries: {
    rating: [],
  },
  advancedQueryHistory: [],
  currentAdvancedFilter: '{}',
  advancedSearchMapVisible: true,
  advancedSearchMapZoom: MAP_DEFAULT_ZOOM,
};

const addQueryHistory = (queryHistory, query) => [].concat([query], queryHistory);

const reducer = createReducer({
  [actionTypes.SET_ADVANCED_SEARCH_QUERY]: (state, action) => {
    const { courses, query } = action;
    return u(
      {
        queries: { [query]: courses },
        advancedQueryHistory: addQueryHistory(
          state.advancedQueryHistory,
          query,
        ),
      },
      state,
    );
  },
  [actionTypes.SET_ADVANCED_FILTER_TYPE]: (state, action) => {
    const { filterName, data } = action;
    return u.updateIn(['advancedQueries', filterName], data);
  },
  [actionTypes.SET_CURRENT_ADVANCED_FILTER]: (state, action) => {
    const { filter } = action;
    return { ...state, currentAdvancedFilter: filter };
  },
  [actionTypes.TOGGLE_ADVANCED_SEARCH_MAP]: (state, action) => {
    const { visible } = action;
    return { ...state, advancedSearchMapVisible: visible };
  },
  [actionTypes.SET_ADVANCED_MAP_ZOOM]: (state, action) => {
    const { zoom } = action;
    return { ...state, advancedSearchMapZoom: zoom };
  },
  [actionTypes.TOGGLE_ADVANCED_SEARCH_INPUTS]: (state, action) => {
    const { open } = action;
    return { ...state, allAdvancedSearchInputsOpen: open };
  },
});

export default reducer;
