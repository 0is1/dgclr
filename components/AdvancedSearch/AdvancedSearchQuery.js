// @flow
import React, { Component } from 'react';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import Link from 'next/link';
import { connect } from 'react-redux';
import { differenceBy, size } from 'lodash';
import { Box } from 'rebass';
import { GoChevronRight } from 'react-icons/go';
import { getRandomKey, uniqueLayoutRatings } from 'helpers/utils';
import { COURSE_QUERY } from 'lib/constants';
import LayoutRatingBadges from 'components/Layout/Badges';
import { ClipLoader } from 'components/Spinners';
import Styles from 'components/SearchContainer/SearchContainer.styles';
import BaseStyles from 'components/Container/Container.styles';
import type { GraphQLData, State } from 'lib/types';
import { setCourses as setCoursesFunc } from 'components/Course/actions';
import { queryResultsFromState as queryResultsFromStateFunc } from 'components/SearchContainer/selectors';
import { setAdvancedSearchQuery } from './actions';
import { latestAdvancedQuery as latestQueryFunc } from './selectors';
import AdvancedSearchQueryStyles from './AdvancedSearchQuery.styles';

const { UL, LI } = BaseStyles;
const { SearchResultItem, SearchResultIcon } = Styles;
const { NoResults } = AdvancedSearchQueryStyles;
type Props = {
  filter: {},
  queryResults: [],
  data: GraphQLData,
  latestQuery: string,
  setCourses: Function,
  setSearchQuery: Function,
};

export class AdvancedSearchQuery extends Component<Props> {
  componentDidUpdate(prevProps, prevState, snapshot) {
    if (snapshot !== null) {
      const { data = {} } = this.props;
      const { courses = [] } = data;
      if (courses) {
        this.setCourses(courses);
      }
    }
  }

  getSnapshotBeforeUpdate(prevProps) {
    const { data = {}, latestQuery, filter } = this.props;
    const { courses = [] } = data;
    const prevCourseData = prevProps.data;
    const prevCourseArr = prevCourseData && prevProps.data.courses;
    const prevCourses = prevCourseArr && prevCourseArr.length ? prevCourseArr : [];
    const stringifyFilter = JSON.stringify(filter);
    // console.log('courses: ', courses);
    // console.log('prevCourses: ', prevCourses);
    if (
      (courses
        && courses.length
        && differenceBy(courses, prevCourses, '_id').length > 0)
      || (!courses.length && stringifyFilter !== latestQuery)
      || (courses.length !== prevCourses.length && stringifyFilter !== latestQuery)
    ) {
      return true;
    }
    return null;
  }

  setCourses = (courses) => {
    const { setCourses, setSearchQuery, filter } = this.props;
    setCourses(courses);
    setSearchQuery(courses, JSON.stringify(filter));
  };

  render() {
    const { filter, queryResults, data = {} } = this.props;
    if (size(filter) < 1 && !queryResults.length) return null;
    if (data && data.loading) {
      return <ClipLoader />;
    }
    const { courses } = data;
    const courseData = queryResults.length ? queryResults : courses;
    let results = false;
    if (courseData && courseData.length) {
      results = courseData.map((course) => {
        const ratings = uniqueLayoutRatings(course.layouts);
        return (
          <LI key={getRandomKey()} listStyle="none">
            <Link as={`/${course.slug}`} href={`/course?slug=${course.slug}`}>
              <SearchResultItem>
                {course.name}
                <LayoutRatingBadges tiny ratings={ratings} />
                <SearchResultIcon>
                  <GoChevronRight size="1.5rem" color="#0067ee" />
                </SearchResultIcon>
              </SearchResultItem>
            </Link>
          </LI>
        );
      });
    }
    if (!results) {
      return (
        <Box p={[0, '0.5rem 2rem']}>
          <NoResults>Ei hakutuloksia!</NoResults>
        </Box>
      );
    }
    return (
      <Box p={[0, '0.5rem 2rem']}>
        <UL>{results}</UL>
      </Box>
    );
  }
}

const mapStateToProps = (state: State) => ({
  latestQuery: latestQueryFunc(state),
  queryResults: queryResultsFromStateFunc(state, latestQueryFunc(state)),
});

const mapDispatchToProps = (dispatch: Function) => ({
  setCourses: courses => dispatch(setCoursesFunc(courses)),
  setSearchQuery: (courses, query) => dispatch(setAdvancedSearchQuery(courses, query)),
});

const SEARCH_COURSES = gql`
  query AdvancedCoursesQuery($filter: CourseQueryFilterInput!) {
    courses(filter: $filter, limit: 150) {
      ${COURSE_QUERY}
    }
  }
`;

// The `graphql` wrapper executes a GraphQL query and makes the results
// available on the `data` prop of the wrapped component (PostList)
const ComponentWithMutation = graphql(SEARCH_COURSES, {
  options: ({ filter }) => ({
    variables: {
      filter,
    },
  }),
  props: ({ data }) => ({
    data,
  }),
  skip: ({ filter }) => !filter || size(filter) < 1,
})(AdvancedSearchQuery);

export default connect<any, Props, any, any, any, Function>(
  mapStateToProps,
  mapDispatchToProps,
)(ComponentWithMutation);
