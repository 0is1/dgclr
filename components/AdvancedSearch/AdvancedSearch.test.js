import React from 'react';
import { shallow } from 'enzyme';
import { AdvancedSearch } from './AdvancedSearch';

const mockProps = (props = {}) => ({
  filter: '{}',
  mapChecked: true,
  ...props,
});

describe('<AdvancedSearch />', () => {
  describe('snapshots', () => {
    it('renders with default props', () => {
      const props = mockProps();
      const subject = shallow(<AdvancedSearch {...props} />);
      expect(subject).toMatchSnapshot();
    });
  });
});
