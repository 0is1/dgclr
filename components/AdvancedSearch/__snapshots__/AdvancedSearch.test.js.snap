// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedSearch /> snapshots renders with default props 1`] = `
<AdvancedSearchstyles__Wrapper>
  <Box
    m="1rem auto"
    px="2rem"
    style={
      Object {
        "position": "relative",
      }
    }
    width={
      Array [
        1,
        1,
        1,
        0.7,
      ]
    }
  >
    <Connect(withI18nextTranslation(AdvancedSearchInputs)) />
    <Connect(Apollo(AdvancedSearchQuery))
      filter={Object {}}
    />
  </Box>
</AdvancedSearchstyles__Wrapper>
`;
